qcodes/data/hdf5_format.py

```diff
@@ -260,6 +260,9 @@ def write_metadata(self, data_set, io_manager=None, location=None, read_first=Tr
     @staticmethod
     def write_dict_to_hdf5(data_dict, entry_point):
         for key, item in data_dict.items():
+            if isinstance(item, (np.int32, np.int64)):
+                item = int(item)
+
             if isinstance(item, (str, bool, tuple, float, int)):
```

Comment:
These are mutually exclusive so this is more efficient.
```suggestion
            elif isinstance(item, (str, bool, tuple, float, int)):
```